posthog/api/property_definition.py

```diff
@@ -68,7 +68,11 @@ def get_queryset(self):
                 use_entreprise_taxonomy = False
 
         properties_to_filter = self.request.GET.get("properties", None)
-        if properties_to_filter:
+        properties_belonging_to_event = self.request.GET.get("event", None)
+        if properties_belonging_to_event:
+            names = tuple(EventProperty.objects.filter(event=properties_belonging_to_event).values_list("property"))
+            name_filter = "AND name IN %(names)s"
+        elif properties_to_filter:
```

Comment:
There's another way to do it... 
https://github.com/PostHog/posthog/pull/8444 adds a filter `is_event_property=true` that only returns properties for this event, without doing such a `values_list` query that potentially moves a lot of data in and out of postgres.

(Insert lecture about always reducing WIP first by doing reviews here... 😅)